src/cargo/ops/cargo_compile.rs

```diff
@@ -110,22 +111,33 @@ impl Packages {
     }
 
     pub fn to_package_id_specs(&self, ws: &Workspace<'_>) -> CargoResult<Vec<PackageIdSpec>> {
-        let specs = match *self {
+        let specs = match self {
             Packages::All => ws
                 .members()
                 .map(Package::package_id)
                 .map(PackageIdSpec::from_package_id)
                 .collect(),
-            Packages::OptOut(ref opt_out) => ws
-                .members()
-                .map(Package::package_id)
-                .map(PackageIdSpec::from_package_id)
-                .filter(|p| opt_out.iter().position(|x| *x == p.name()).is_none())
-                .collect(),
-            Packages::Packages(ref packages) if packages.is_empty() => {
+            Packages::OptOut(opt_out) => {
+                let mut opt_out: HashSet<&String> = HashSet::from_iter(opt_out);
+                let packages = ws
+                    .members()
+                    .filter(|pkg| !opt_out.remove(&pkg.name().as_str().to_string()))
+                    .map(Package::package_id)
+                    .map(PackageIdSpec::from_package_id)
+                    .collect();
+                if !opt_out.is_empty() {
+                    ws.config().shell().warn(format!(
+                        "excluded package(s) {:?} not found in workspace `{}`",
```

Comment:
I think we may want to avoid `:?` here as it's not quite suitable for display to end users

Reply:
Sure, I'll try something else.  It's just annoying to not have something convenient in the library (or not have a dependency on itertools).